### PR TITLE
Add UniversalFallback

### LINKED ISSUE #285: Instance should support arbitrary attributes
Re-opening from https://github.com/JuliaOpt/MathOptInterfaceBridges.jl/issues/61. I need a solution to this for NLP to work.

### PR DESCRIPTION
**Pull request by blegat**
Closes #285

### CODE REVIEW

File: src/Utilities/universalfallback.jl [hunk 1]
```diff
@@ -0,0 +1,109 @@
+"""
+    UniversalFallback
+
+The `UniversalFallback` can be applied on a `ModelLike` `model` to create the model `UniversalFallback(model)` supporting *any* constraints and attributes.
+This allows to have specialized implementation in `model` for performance critical constraints and attributes while still supporting other constraints and attributes with a small performance penalty.
```
Comment (mlubin) [inline src/Utilities/universalfallback.jl]: *a* specialized implementation

File: src/Utilities/universalfallback.jl [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                           Dict{MOI.AbstractModelAttribute, Any}(),
+                           Dict{MOI.AbstractVariableAttribute, Dict{VI, Any}}(),
+                           Dict{MOI.AbstractConstraintAttribute, Dict{CI, Any}}())
+    end
+end
+UniversalFallback(model::MOI.ModelLike) = UniversalFallback{typeof(model)}(model)
+
+MOI.isempty(uf::UniversalFallback) = MOI.isempty(uf.model)
+MOI.empty!(uf::UniversalFallback) = MOI.empty!(uf.model)
+MOI.copy!(uf::UniversalFallback, src::MOI.ModelLike) = MOIU.defaultcopy!(b, src)
```
Comment (mlubin) [inline src/Utilities/universalfallback.jl]: what's `b`? (add test coverage)

File: src/Utilities/universalfallback.jl [hunk 3]
```diff
@@ -0,0 +1,109 @@
+"""
+    UniversalFallback
+
+The `UniversalFallback` can be applied on a `ModelLike` `model` to create the model `UniversalFallback(model)` supporting *any* constraints and attributes.
```
Comment (mlubin) [inline src/Utilities/universalfallback.jl]: Does this actually support any constraint? It looks like it just forwards all constraints to the model.

Comment (blegat) [reply src/Utilities/universalfallback.jl]: Yes, currently I have only attributes fallback but I was planning to add constraints too to fix https://github.com/JuliaOpt/JuMP.jl/issues/1152

Comment (mlubin) [reply src/Utilities/universalfallback.jl]: Having constraints working will be great :)
Just update the comment so it accurately describes the code that exists right now.

**Pull request merged**